### PR TITLE
Fix build error on MinGW/MSYS environment

### PR DESCRIPTION
**Pull request by msakai**
This PR fixes build error on MinGW/MSYS environment with CMake's "MSYS Makefiles" generator.

I'm not sure whether supporting MinGW is worthwhile, but the changes are small enough and I guess it does not adversely affect other platform.

### CODE REVIEW

**Comment by rsdubtso** [comment]
The only concern that I have about this is that we definitely are not going to do testing on MinGW ever. So things may break down the road without anybody noticing. 

Related question: does code compiled with MinGW use Unix AMD64 ABI or Windows x64 ABI?

**Comment by msakai** [comment]
> The only concern that I have about this is that we definitely are not going to do testing on MinGW ever. So things may break down the road without anybody noticing.

I understand.
MinGW is handy for me (and possibly some other developers) than MSVC so that I feel it would be nice if the PR is merged. But I think you don't need to support MinGW officially, and it would be okay even if it sometimes does not work.

Also, maybe I can try to write AppVeyor configuration for testing MinGW build continuously.

> Related question: does code compiled with MinGW use Unix AMD64 ABI or Windows x64 ABI?

I do not know much about ABI, but MinGW's is compatible with MSVC for C level functions.
So I suppose it uses Windows x64 ABI.

File: src/cpu/jit_avx512_common_gemm_f32.cpp [hunk 1]
```diff
@@ -63,7 +63,7 @@ struct jit_avx512_common_gemm_f32::xbyak_gemm : public jit_generator {
         auto ARG_N = abi_param2;
         auto K = abi_param3;
         auto ARG_ALPHA = abi_param4;
-#ifdef _WIN32
+#if defined(WIN32) && !defined(__GNUC__)
```
Comment (rsdubtso) [inline src/cpu/jit_avx512_common_gemm_f32.cpp]: I'm not sure I understand this part. If MinGW uses Windows x64 ABI, then we need to generate the same prologue regardless of the compiler.

Comment (msakai) [reply src/cpu/jit_avx512_common_gemm_f32.cpp]: Thank you for pointed out.I think you're right.

I was careless on that.
After I accidentally undefined  `OFFSET_SHADOWSPACE`, I just chose the code just passed the compilation without thinking properly.

I'll add a commit for properly fixing the issue.

**Comment by emfomenk** [comment]
Hi @msakai,

I checked with MSYS2 and everything works just fine.
Looks good to me as well.
Thanks a lot for the contribution!

**Pull request merged**